sycl/test-e2e/lit.cfg.py

```diff
@@ -330,29 +330,32 @@ def open_check_file(file_name):
 else:
     config.substitutions.append(("%level_zero_options", ""))
 
-# Check for sycl-preview library
-check_preview_breaking_changes_file = "preview_breaking_changes_link.cpp"
-with open_check_file(check_preview_breaking_changes_file) as fp:
-    print(
-        textwrap.dedent(
+if lit_config.params.get("test-preview-mode", False):
+    config.available_features.add("preview-mode")
+else:
+    # Check for sycl-preview library
+    check_preview_breaking_changes_file = "preview_breaking_changes_link.cpp"
+    with open_check_file(check_preview_breaking_changes_file) as fp:
+        print(
+            textwrap.dedent(
+                """
+            #include <sycl/sycl.hpp>
+            namespace sycl { inline namespace _V1 { namespace detail {
+            extern void PreviewMajorReleaseMarker();
+            }}}
+            int main() { sycl::detail::PreviewMajorReleaseMarker(); return 0; }
             """
-        #include <sycl/sycl.hpp>
-        namespace sycl { inline namespace _V1 { namespace detail {
-        extern void PreviewMajorReleaseMarker();
-        }}}
-        int main() { sycl::detail::PreviewMajorReleaseMarker(); return 0; }
-        """
-        ),
-        file=fp,
-    )
+            ),
+            file=fp,
+        )
 
-sp = subprocess.getstatusoutput(
-    config.dpcpp_compiler
-    + " -fsycl -fpreview-breaking-changes "
-    + check_preview_breaking_changes_file
-)
-if sp[0] == 0:
-    config.available_features.add("preview-breaking-changes-supported")
+    sp = subprocess.getstatusoutput(
+        config.dpcpp_compiler
+        + " -fsycl -fpreview-breaking-changes "
+        + check_preview_breaking_changes_file
+    )
+    if sp[0] == 0:
+        config.available_features.add("preview-breaking-changes-supported")
```

Comment:
What's the difference between `preview-mode` and `preview-breaking-changes-supported` LIT features?

Reply:
One is "can we use toolchain to compile tiny pieces in preview mode?". Another is "run **everything** in preview mode". The way we use the former is to add some testing targeting preview functionality. The problem with this approach is that we do very little regression testing and we can't leverage existing tests without purposefully modifying them. The latter would allow us to catch regressions much earlier, unlike the previous breaking window when some bits were only caught up when making the final transition.